crates/node/service/src/actors/engine.rs

```diff
@@ -104,12 +119,38 @@ impl NodeActor for EngineActor {
                         warn!(target: "engine", "Encountered error draining engine api tasks: {:?}", e);
                     }
                 }
+                attributes = self.attributes_rx.recv() => {
+                    let Some(attributes) = attributes else {
+                        debug!(target: "engine", "Received `None` attributes from receiver");
+                        continue;
+                    };
+                    // Skip to processing the payload attributes if consolidation is not needed.
+                    if !self.engine.needs_consolidation() {
+                        let task = BuildTask::new(
+                            Arc::clone(&self.client),
+                            Arc::clone(&self.config),
+                            attributes.clone(),
+                            true,
+                        );
+                        let task = EngineTask::BuildBlock(task);
+                        self.engine.enqueue(task);
+                    } else {
+                        let task = ConsolidateTask::new(
+                            Arc::clone(&self.client),
+                            Arc::clone(&self.config),
+                            attributes,
+                            true,
+                        );
+                        let task = EngineTask::Consolidate(task);
+                        self.engine.enqueue(task);
+                    }
```

Comment:
Your consolidate task looks to have this branch contained within it - it returns early with just a FCU if the match checks out. Maybe we can remove `needs_consolidation` and just leave that up to the `ConsolidateTask` to handle?

Reply:
Yep that's a perfectly fine route to go - probably nicer to head the direction of keeping logic encapsulated within the tasks rather than leaking out into the actor